src/alloc/isolated_alloc.rs

```diff
@@ -60,12 +60,13 @@ impl IsolatedAlloc {
 
     /// If the allocation is greater than a page, then round to the nearest page #.
     #[inline]
-    fn huge_normalized_layout(layout: Layout, page_size: usize) -> Layout {
+    fn huge_normalized_layout(&self, layout: Layout, page_size: usize) -> (Layout, usize) {
```

Comment:
This is odd, since the resulting `Layout` is not what we are actually using -- we can't even tell `mmap` an alignment.

So really the "number of pages to allocate" is the only thing this should return I think.

Reply:
Made it return the size directly instead of page count since that's what we use, but good point. Also got rid of the page_layout function since we only care about the size when mmapping